### PR TITLE
Use `debug` modules to parse Go version and modules

### PR DESCRIPTION
**Pull request by damemi**
Fixes https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/652

### CODE REVIEW

**Review by MrAlias — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ OpenTelemetry Go Automatic Instrumentation adheres to [Semantic Versioning](http
 
 - Don't call `manager.Close()` when `Analyze()` fails. ([#638](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/638))
 - Close `proc` file when done discovering PID. ([#649](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/649))
+- Use `debug` modules to parse Go version and modules. ([#653](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/653))
```
Comment (pellared) [inline CHANGELOG.md:15]: nit

```suggestion
- Use `debug` packages to parse Go and modules' versions. ([#653](https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/653))
```

File: internal/pkg/process/analyze.go [hunk 2]
```diff
@@ -80,15 +82,24 @@ func (a *Analyzer) Analyze(pid int, relevantFuncs map[string]interface{}) (*Targ
...
 	result.GoVersion = goVersion
-
-	// Include the Go standard library module.
-	modules["std"] = goVersion
-	result.Libraries = modules
+	result.Libraries = make(map[string]*version.Version)
+	for _, dep := range buildInfo.Deps {
+		depVersion, err := version.NewVersion(dep.Version)
+		if err != nil {
+			return nil, err
```
Comment (RonFed) [inline internal/pkg/process/analyze.go:98]: Not sure if we should return an error here. The current behavior is to log the error and continue. If I remember correctly, when parsing version with commit hashes in them, the parsing might fail.

File: internal/pkg/process/analyze.go [hunk 3]
```diff
@@ -80,15 +82,24 @@ func (a *Analyzer) Analyze(pid int, relevantFuncs map[string]interface{}) (*Targ
...
+	goVersion, err := version.NewVersion(strings.ReplaceAll(buildInfo.GoVersion, "go", ""))
 	if err != nil {
 		return nil, err
 	}
 	result.GoVersion = goVersion
-
-	// Include the Go standard library module.
-	modules["std"] = goVersion
-	result.Libraries = modules
+	result.Libraries = make(map[string]*version.Version)
```
Comment (pellared) [inline internal/pkg/process/analyze.go:94]: nit

```suggestion
	result.Libraries = make(map[string]*version.Version, len(buildInfo.Deps) + 1)
```

**Review by pellared — APPROVED** [review]

File: internal/pkg/process/analyze.go (hunk 2 above)
Comment (damemi) [reply internal/pkg/process/analyze.go]: done

File: CHANGELOG.md (hunk 1 above)
Comment (damemi) [reply CHANGELOG.md]: done

File: internal/pkg/process/analyze.go (hunk 3 above)
Comment (damemi) [reply internal/pkg/process/analyze.go]: done

**Pull request merged**